R/dataset-stream.R

```diff
@@ -0,0 +1,62 @@
+#' Get the pending streams for a dataset
+#'
+#' Crunch allows you to stream data to a dataset. Streaming data is useful for
+#' datasets which have frequent updates (see 
+#' [the Crunch documentation](http://docs.crunch.io/#streaming-rows) for more)
+#' information. `pendingMessages()` retrieves the number of pending messages and `appendStreamedRows()` appends all pending streamed rows to the dataset.
+#'
+#' @param ds a CrunchDataset
+#' @return number of pending messages in the stream for the dataset
+#' @export
+pendingStream <- function (ds) {
+    stopifnot(is.dataset(ds))
+    stream_cat <- ShojiEntity(crGET(shojiURL(ds, "fragments", "stream")))
+    stream_cat$pending_messages
+}
+
+#' Stream data to a Crunch dataset
+#' 
+#' @param ds a CrunchDataset
+#' @param data a dataframe with data to send as a stream, The given data values
+#'  must be in the Crunch I/O format (for example, category ids instead of 
+#'  names or numeric_values)
+#' @keywords internal
+streamRows <- function (ds, data) {
+    if (nrow(data) > 1) {
+        payload <- by(data, seq_len(nrow(data)), function (row) toJSON(row))
+        payload <- paste0(payload, collapse = "\n") 
+        crPOST(shojiURL(ds, "fragments", "stream"), body=payload)
+    }
+    invisible(refresh(ds))
+}
+
+#' Append data that has been streamed to a dataset
+#'
+#' Crunch allows you to stream data to a dataset. Streaming data is useful for
+#' datasets which have frequent updates (see 
+#' [the Crunch documentation](http://docs.crunch.io/#streaming-rows) for more)
+#' information. Crunch automatically appends streamed data perioadically,
+#' however if you would like to trigger appending pending streamed data to a 
+#' dataset you can use `appendStream()`.
+#'
+#' @param ds a CrunchDataset
+#' @param ... Additional batch attributes to set when appending pending stream 
```

Comment:
Nah, you can remove this. No one names batches. Easy to add if that need somehow arises, but until them let's keep it simple.